packages/cli/src/lib/webpack/run-webpack.js

```diff
@@ -69,100 +65,81 @@ async function prodBuild(config, env) {
 	const clientWebpackConfig = await clientConfig(config, env);
 	await transformConfig(clientWebpackConfig, config, env);
 	const clientCompiler = webpack(clientWebpackConfig);
+	await runCompiler(clientCompiler);
 
-	try {
-		let stats = await runCompiler(clientCompiler);
-
-		// Timeout for plugins that work on `after-emit` event of webpack
-		await new Promise(r => setTimeout(r, 20));
-
-		return showStats(stats, true);
-	} catch (err) {
-		// eslint-disable-next-line
-		console.log(err);
-	}
+	// Timeout for plugins that work on `after-emit` event of webpack
+	await new Promise(r => setTimeout(r, 20));
 }
 
+/**
+ * @param {import('webpack').Compiler} compiler
+ */
 function runCompiler(compiler) {
-	return new Promise(res => {
+	return new Promise((res, rej) => {
 		compiler.run((err, stats) => {
-			if (err) {
-				error(err, 1);
-			}
+			if (err) rej(err);
 
-			showStats(stats, true);
+			showCompilationIssues(stats);
 
-			res(stats);
+			compiler.close(closeErr => {
+				if (closeErr) rej(closeErr);
+				res();
+			});
 		});
 	});
 }
 
-function showStats(stats, isProd) {
+/**
+ * @param {import('webpack').Stats} stats
+ */
+function showCompilationIssues(stats) {
 	if (stats) {
-		if (stats.hasErrors()) {
-			allFields(stats, 'errors')
-				.map(stripLoaderPrefix)
-				.forEach(({ message }) => error(message, isProd ? 1 : 0));
+		if (stats.hasWarnings()) {
+			allFields(stats, 'warnings').forEach(({ message }) => warn(message));
 		}
 
-		if (stats.hasWarnings()) {
-			allFields(stats, 'warnings')
-				.map(stripLoaderPrefix)
-				.forEach(({ message }) => warn(message));
+		if (stats.hasErrors()) {
+			allFields(stats, 'errors').forEach(err => {
+				throw err;
+			});
```

Comment:
I switched the order of warning & error reporting here; as we bail on errors, a user would need an error-less build in order to be alerted of any warnings which I think is less than ideal.

If they're alerted to warnings first, they can fix those alongside whatever error they're experiencing.